composer.json

```diff
@@ -10,22 +10,35 @@
         {"name": "Roman Borschel", "email": "roman@code-factory.org"},
         {"name": "Benjamin Eberlei", "email": "kontakt@beberlei.de"},
         {"name": "Jonathan Wage", "email": "jonwage@gmail.com"},
-        {"name": "Johannes Schmitt", "email": "schmittjoh@gmail.com"}
+        {"name": "Johannes Schmitt", "email": "schmittjoh@gmail.com"},
+        {"name": "Fabio B. Silva", "email": "fabio.bat.silva@gmail.com"}
     ],
     "require": {
-        "php": ">=5.3.2",
-        "doctrine/lexer": "1.*"
+        "php":          ">=7.0.0",
+        "hoa/compiler": "3.*"
     },
     "require-dev": {
-        "doctrine/cache": "1.*",
-        "phpunit/phpunit": "4.*"
+        "doctrine/cache"                        : "1.*",
+        "phpunit/phpunit"                       : "5.*",
+        "instaclick/coding-standard"            : "~1.1",
+        "squizlabs/php_codesniffer"             : "~1.5",
+        "satooshi/php-coveralls"                : "~1.0",
+        "instaclick/object-calisthenics-sniffs" : "dev-master",
+        "instaclick/symfony2-coding-standard"   : "dev-remaster"
     },
     "autoload": {
-        "psr-0": { "Doctrine\\Common\\Annotations\\": "lib/" }
+        "psr-4": {
+            "Doctrine\\Annotations\\": "src/"
```

Comment:
Folks, please bring the deps discussion into a new issue, if necessary.

While it is true that a major bump will cause issues for dependant
packages, we will deal with polyfilling once the cleanup by @FabioBatSilva
is done.

Focus on the design problems inside the patch, for now, and focus on what
you'd like the package to be, as if it had no dependents at all.

In composer.json
<https://github.com/doctrine/annotations/pull/75#discussion_r57308250>:

>      },
>      "autoload": {
> -        "psr-0": { "Doctrine\\Common\\Annotations\\": "lib/" }
> +        "psr-4": {
> +            "Doctrine\\Annotations\\": "src/"

Don't get me wrong @stof <https://github.com/stof>, I'm just saying that we
shouldn't maintain a new library for deprecated PHP version only to follow
the Symfony requirements.

This is not what I asked

What is the "best" time to reorder namespaces, if not when we introduce a
new major version ? Symfony did it too, even before Symfony 3 (for Form
component between 2.4 and 2.5, introducing a new API).

The best way to make a class renaming is to have a minor version supporting
both names, and then removing the deprecated class in the next major. this
would allow supporting both major at the same time (as the new API would be
available in latest versions of the previous major). Please read the
article I linked above. This is precisely what it explains.

Also, what is the gain of this namespace renaming here ?

This is the real question. AFAIK, there is no benefit (except making it
obvious that this class is not in doctrine/common itself anymore, but
installing doctrine/common installs doctrine/annotations`too anyway) and
only drawbacks.

Annotations is quite stable, we can continue to maintain the 1.x branch by
doing almost nothing :-)

That's not the only question. Making it very hard to support both majors at
the same time (again, the only thing making it hard here is the namespace
renaming) means that you expect the whole PHP community to upgrade their
packages to doctrine/annotations 2.x at the same time (and this can only be
done after they all drop support of PHP 5.x, which means it won't happen
soon).
My whole point here is to make incremental migrations possible by allowing
other packages to depend on ^1.3 || ^2.0 for most of them.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub
<https://github.com/doctrine/annotations/pull/75/files/393ca1de927f97ecb4ae55a2c2e780e2b9fd5e0d#r57308250>